include/boost/geometry/algorithms/visit.hpp

```diff
@@ -0,0 +1,307 @@
+        "Not implemented for these types.",
+        Geometry1, Geometry2);
+};
+
+template <typename Geometry1, typename Geometry2, typename Tag2>
+struct visit_two<Geometry1, Geometry2, dynamic_geometry_tag, Tag2>
+{
+    template <typename F, typename G1, typename G2>
+    static void apply(F && f, G1 & geom1, G2 & geom2)
+    {
+        traits::visit<util::remove_cref_t<G1>>::apply([&](auto & g1)
+        {
+            f(g1, geom2);
+        }, geom1);
+    }
+};
+
+template <typename Geometry1, typename Geometry2, typename Tag1>
+struct visit_two<Geometry1, Geometry2, Tag1, dynamic_geometry_tag>
+{
+    template <typename F, typename G1, typename G2>
+    static void apply(F && f, G1 & geom1, G2 & geom2)
+    {
+        traits::visit<util::remove_cref_t<G2>>::apply([&](auto & g2)
+        {
+            f(geom1, g2);
+        }, geom2);
+    }
+};
+
+template <typename Geometry1, typename Geometry2>
+struct visit_two<Geometry1, Geometry2, dynamic_geometry_tag, dynamic_geometry_tag>
+{
+    template <typename F, typename G1, typename G2>
+    static void apply(F && f, G1 & geom1, G2 & geom2)
+    {
+        traits::visit
+            <
+                util::remove_cref_t<G1>, util::remove_cref_t<G2>
+            >::apply([&](auto & g1, auto & g2)
+            {
+                f(g1, g2);
+            }, geom1, geom2);
+    }
+};
+
+
+template <typename Geometry, typename Tag = typename tag<Geometry>::type>
+struct visit_breadth_first
+{
+    template <typename F, typename G>
+    static bool apply(F f, G & g)
+    {
+        return f(g);
+    }
+};
+
+template <typename Geometry>
+struct visit_breadth_first<Geometry, void>
+{
+    BOOST_GEOMETRY_STATIC_ASSERT_FALSE(
+        "Not implemented for this Geometry type.",
+        Geometry);
+};
+
+template <typename Geometry>
+struct visit_breadth_first<Geometry, dynamic_geometry_tag>
+{
+    template <typename Geom, typename F>
+    static bool apply(F function, Geom & geom)
+    {
+        bool result = true;
+        traits::visit<util::remove_cref_t<Geom>>::apply([&](auto & g)
+        {
+            result = visit_breadth_first<decltype(g)>::apply(function, g);
+        }, geom);
+        return result;
+    }
+};
+
+template <typename Geometry>
+struct visit_breadth_first<Geometry, geometry_collection_tag>
+{
+    template <typename F, typename Geom>
+    static bool apply(F function, Geom & geom)
+    {
+        using iter_t = typename boost::range_iterator<Geom>::type;
+        std::deque<iter_t> queue;
+
+        iter_t it = boost::begin(geom);
+        iter_t end = boost::end(geom);
+        for(;;)
+        {
+            for (; it != end; ++it)
+            {
+                bool result = true;
+                traits::visit_iterator<util::remove_cref_t<Geom>>::apply([&](auto & g)
+                {
+                    result = visit_or_enqueue(function, g, queue, it);
+                }, it);
+                if (! result)
+                    return false;
```

Comment:
Curly braces are missing

Reply:
What do you think about relaxing this guideline? I.e. to allow single lines under if-else? From my point of view adding curly braces around this one line makes the whole algorithm less readable because there are 2x as much lines of code that can be seen by the eye, the information is the same and this space could be used for other things.

It'd be a different thing if this was very long line which potentially could be wrapped into yet another line. In this case I'd argue that curly braces are indeed desireable. But in case of a short call or return statement. I don't think that it is.

Do you strongly feel that these braces are necessary here?

Reply:
I added them but I'll probably create an issue to discuss this later.